Assets/Scripts/DeadEnemy.cs

```diff
@@ -0,0 +1,24 @@
+﻿using System.Collections;
+using System.Collections.Generic;
+using UnityEngine;
+
+public class DeadEnemy : MonoBehaviour
+{
+    public int health = -1;
+    public GameObject revivedEnemy;
+
+    void Update()
+    {
+        if (health >= 0)
+        {
+            Destroy(gameObject);
```

Comment:
It's risky to destroy the gameobject this script is on before you are done executing all of the code you need to call in the script. I.E. make Destroy the last thing.